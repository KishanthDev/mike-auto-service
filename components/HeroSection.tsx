import { Button } from "@heroui/button";
import React from "react";

const HeroSection = () => {
  return (
    <section className="h-[300px] md:h-[400px] bg-cover bg-center bg-[url('/api/placeholder/1200/300')] relative flex flex-col justify-center items-center text-white text-center px-4 dark:text-gray-200">
      <div className="bg-black bg-opacity-50 absolute inset-0 z-0" />
      <div className="relative z-10 max-w-4xl px-4">
        <h1 className="text-3xl sm:text-4xl md:text-5xl font-bold mb-2 md:mb-4">Mike's Auto Service</h1>
        <p className="text-base sm:text-lg md:text-xl max-w-[800px]">
          Full-service auto repair shop specializing in domestic and foreign vehicles since 1985
        </p>
        <div className="flex flex-col sm:flex-row justify-center items-center mt-4 md:mt-6 gap-2">
          <span className="text-yellow-400 text-2xl mr-2">★★★★★</span>
          <span>4.8/5 based on 127 reviews</span>
        </div>
        <div className="flex flex-col sm:flex-row gap-4 mt-6 justify-center">
          <Button className="bg-red-600 text-white w-full sm:w-auto">Schedule Service</Button>
          <Button className="bg-white text-[#333] dark:bg-gray-100 dark:text-black w-full sm:w-auto">Request Quote</Button>
        </div>
      </div>
    </section>
  );
};

export default HeroSection;