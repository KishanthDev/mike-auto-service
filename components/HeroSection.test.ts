import React from "react";
import { render, screen } from "@testing-library/react";
import HeroSection from "./HeroSection";

// Mock the Button component using React.createElement
jest.mock("@heroui/button", () => ({
  Button: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) => React.createElement("button", { className }, children),
}));

describe("HeroSection Component", () => {
  beforeEach(() => {
    render(React.createElement(HeroSection));
  });

  it("displays the correct heading", () => {
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Mike's Auto Service");
    expect(heading).toHaveClass("text-3xl");
  });

  it("displays the correct description text", () => {
    const description = screen.getByText(
      /Full-service auto repair shop specializing in domestic and foreign vehicles since 1985/i,
    );
    expect(description).toBeInTheDocument();
    expect(description).toHaveClass("text-base");
  });

  it("displays the correct rating information", () => {
    const stars = screen.getByText("★★★★★");
    const ratingText = screen.getByText(/4.8\/5 based on 127 reviews/i);

    expect(stars).toBeInTheDocument();
    expect(stars).toHaveClass("text-yellow-400");
    expect(ratingText).toBeInTheDocument();
  });

  describe("Buttons", () => {
    it("renders both buttons", () => {
      const buttons = screen.getAllByRole("button");
      expect(buttons).toHaveLength(2);
    });

    it("has a Schedule Service button with correct styling", () => {
      const scheduleButton = screen.getByText("Schedule Service");
      expect(scheduleButton).toBeInTheDocument();
      expect(scheduleButton).toHaveClass("bg-red-600");
    });

    it("has a Request Quote button with correct styling", () => {
      const quoteButton = screen.getByText("Request Quote");
      expect(quoteButton).toBeInTheDocument();
      expect(quoteButton).toHaveClass("bg-white");
    });
  });

  it("has correct overlay styling", () => {
    const overlay = screen.getByTestId("hero-overlay");
    expect(overlay).toBeInTheDocument();
    expect(overlay).toHaveClass("bg-black");
    expect(overlay).toHaveClass("bg-opacity-50");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(React.createElement(HeroSection));
    expect(asFragment()).toMatchSnapshot();
  });
});
