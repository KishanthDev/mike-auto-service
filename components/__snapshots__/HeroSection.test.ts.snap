// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="h-[400px] md:h-[400px] bg-cover bg-center relative flex flex-col justify-center items-center text-white text-center p-4 dark:text-gray-200"
  >
    <div
      class="bg-black bg-opacity-50 absolute inset-0 z-0"
      data-testid="hero-overlay"
    />
    <div
      class="relative z-10 max-w-4xl px-4"
    >
      <h1
        class="text-3xl sm:text-4xl md:text-5xl font-bold mb-2 md:mb-4"
      >
        Mike's Auto Service
      </h1>
      <p
        class="text-base sm:text-lg md:text-xl max-w-[800px]"
      >
        Full-service auto repair shop specializing in domestic and foreign vehicles since 1985
      </p>
      <div
        class="flex flex-col sm:flex-row justify-center items-center mt-4 md:mt-6 gap-2"
      >
        <span
          class="text-yellow-400 text-2xl mr-2"
        >
          ★★★★★
        </span>
        <span>
          4.8/5 based on 127 reviews
        </span>
      </div>
      <div
        class="flex flex-col sm:flex-row gap-4 mt-6 justify-center"
      >
        <button
          class="bg-red-600 text-white w-full sm:w-auto"
        >
          Schedule Service
        </button>
        <button
          class="bg-white text-[#333] dark:bg-gray-100 dark:text-black w-full sm:w-auto"
        >
          Request Quote
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
